scripts/lint.rust.sh

```diff
@@ -0,0 +1,4 @@
+#!/usr/bin/env bash
+set -euo pipefail
+# Lint the rust code
+cargo clippy --locked --target wasm32-unknown-unknown --all-features -- -D warnings -W clippy::pedantic -A clippy::module-name-repetitions -A clippy::struct-field-names
```

Comment:
Note:
- This is a similar set of flags as the nns-dapp.
- The `ic` repo uses `clippy:all` and a forbidden methods file.